packages/core/admin/admin/src/content-manager/hooks/useRelation/useRelation.js

```diff
@@ -81,6 +90,37 @@ export const useRelation = (cacheKey, { relation, search }) => {
     }),
   });
 
+  const { pageGoal } = relation;
+
+  const { status, data, fetchNextPage, hasNextPage } = relationsRes;
+
+  useEffect(() => {
+    /**
+     * This ensures the infiniteQuery hook fetching has caught-up with the modifiedData
+     * state i.e. in circumstances where you add 10 relations, the browserState knows this,
+     * but the hook would think it could fetch more, when in reality, it can't.
+     */
+    if (pageGoal > currentPage && hasNextPage && status === 'success') {
+      fetchNextPage({
+        pageParam: currentPage + 1,
+      });
+    }
+  }, [pageGoal, currentPage, fetchNextPage, hasNextPage, status]);
+
+  useEffect(() => {
+    if (status === 'success' && data && data.pages?.at(-1)?.results && onLoadRelationsCallback) {
+      // everytime we fetch, we normalize prior to adding to redux
+      const normalizedResults = normalizeRelations(data.pages?.at(-1)?.results, normalizeArguments);
+
+      // this is loadRelation from EditViewDataManagerProvider
+      onLoadRelationsCallback({
+        target: { name, value: normalizedResults },
+      });
+    }
```

Comment:
What benefit do you see to handle this as part of `useRelation()`? I'm wondering if the hook wouldn't be easier to use without `normalizeArguments` and `onLoadRelationsCallback`.

Reply:
I think it's better to have it here and run a callback because we never want the relations to not be normalised, a separation of concerns in the sense that `useRelations` should handle everything to do with them, it shouldn't know about the store _but_ it should fire a callback. Otherwise the effect is in the data manager.

I think I _wanted_ to stop returning the `relationsRes.data` because it's not used anyway, we only want the data to be in the store, hence the callback.